homework-solutions/ts/hw-1/hw-3/task-1.ts

```diff
@@ -0,0 +1,55 @@
+// 1. Создайте интерфейс IVehicle:
+//   Методы:
+//     - getDetails(): string — возвращает информацию о транспортном средстве.
+//     - start(): string — возвращает строку "The vehicle is starting".
+
+interface IVehicle {
+  getDetails(): string;
+  start(): string;
+}
+
+// 2. Создайте абстрактный класс Vehicle, который имплементирует IVehicle:
+//   Реализуйте конструктор с полями:
+//     - make (строка)
+//     - model (строка)
+//   Добавьте методы:
+//     - start, возвращающего строку: "The vehicle {make} {model} is starting.".
+//     - Абстрактный метод getDetails, который нужно реализовать в классах-наследниках.
+
+abstract class Vehicle implements IVehicle {
+  constructor(
+    public make: string,
+    public model: string,
+  ) {
+    this.make = make;
+    this.model = model;
+  }
```

Comment:
Если используем модификаторьі доступа в параметрах конструктора, переданьіе аргументьі будут автоматически инициализированьі
```suggestion
    public make: string,
    public model: string,
  ) {}
```